pkg/operator/kube_cloud_config/aws.go

```diff
@@ -0,0 +1,102 @@
+package kube_cloud_config
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-config-operator/pkg/operator/kube_cloud_config/internal/aws"
+	"gopkg.in/gcfg.v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// awsTransformer implements the cloudConfigTransformer. It uses the input ConfigMap and infra.status.platformStatus.AWS.ServiceEndpoints
+// to create a new config that include the ServiceOverrides sections. The transformer uses infra.status.platformStatus.region as the
+// signing service for all the ServiceOverrides.
+// It returns an error if the platform is AWSPlatformType.
+func awsTransformer(input *corev1.ConfigMap, key string, infra *configv1.Infrastructure) (*corev1.ConfigMap, error) {
+	if !(infra.Status.PlatformStatus != nil &&
+		infra.Status.PlatformStatus.Type == configv1.AWSPlatformType) {
+		return nil, fmt.Errorf("invalid platform, expected to be AWS")
+	}
+	if !(infra.Status.PlatformStatus.AWS != nil && len(infra.Status.PlatformStatus.AWS.ServiceEndpoints) > 0) {
+		return asIsTransformer(input, key, infra) // no transformation required
+	}
+
+	var region string
+	if infra.Status.PlatformStatus != nil && infra.Status.PlatformStatus.AWS != nil {
+		region = infra.Status.PlatformStatus.AWS.Region
+	}
+	if region == "" {
+		return nil, fmt.Errorf("Required region for AWS platform infrastructures.%s/cluster not found", configv1.GroupName)
+	}
+
+	output := input.DeepCopy()
+
+	inCfgRaw := &bytes.Buffer{}
+	if v, ok := input.Data[key]; ok {
+		inCfgRaw = bytes.NewBufferString(v)
+	} else if v, ok := input.BinaryData[key]; ok {
+		inCfgRaw = bytes.NewBuffer(v)
+	}
+
+	if len(inCfgRaw.String()) > 0 {
+		var cfg aws.CloudConfig
+		err := gcfg.ReadInto(&cfg, bytes.NewBufferString(inCfgRaw.String()))
+		if err != nil {
+			return nil, fmt.Errorf("failed to read the cloud.conf: %w", err)
+		}
+
+		if len(cfg.ServiceOverride) > 0 {
```

Comment:
What is the downside to checking to see if the values conflict?  A cluster-admin may well have an existing config file that already specifies all the same information, right?

Reply:
because I think it's safer for us to ensure that there is only one way to configure a thing. Allowing user to service override and also in spec means there is duplication of the same information.

Also since there is no marshaller for gcfg _the format used for this configuration file upstream_, there is no sane way to update user specified to match the correct standards in case there is pre-existing.